code/modules/events/spacevine.dm

```diff
@@ -554,24 +560,24 @@
 		to_chat(V, "<span class='danger'>The vines [pick("wind", "tangle", "tighten")] around you!</span>")
 		buckle_mob(V, 1)
 
-/obj/structure/spacevine/proc/spread()
-	var/direction = pick(GLOB.cardinal)
+/obj/structure/spacevine/proc/spread(list/dirs)
+	var/direction = pick(dirs)
 	var/turf/stepturf = get_step(src,direction)
-	for(var/datum/spacevine_mutation/SM in mutations)
-		SM.on_spread(src, stepturf)
-		stepturf = get_step(src,direction) //in case turf changes, to make sure no runtimes happen
-	if(!locate(/obj/structure/spacevine, stepturf))
-		if(stepturf.Enter(src))
-			if(master)
-				master.spawn_spacevine_piece(stepturf, src)
+	if(stepturf)
```

Comment:
You need to use the helpers, to see if the Z levels are connected (see the helpers file in the multiz folder)